gslib/tests/test_update.py

```diff
@@ -84,13 +84,12 @@ def test_update(self):
     # working files left in top-level directory by gsutil developers (like tags,
     # .git*, etc.)
     os.makedirs(gsutil_dst)
-    for comp in ('CHANGES.md', 'CHECKSUM', 'gslib', 'gsutil', 'gsutil.py',
-                 'LICENSE', 'MANIFEST.in', 'README.md', 'setup.py',
-                 'third_party', 'VERSION'):
-      cp_src_path = os.path.join(GSUTIL_DIR, comp)
-      cp_dst_path = os.path.join(gsutil_dst, comp)
-      func = shutil.copytree if os.path.isdir(cp_src_path) else shutil.copyfile
-      func(cp_src_path, cp_dst_path)
+    for comp in os.listdir(GSUTIL_DIR):
```

Comment:
It seems like we should add something to gslib/utils/update_util.py to get the list of expected files for an update (as parsed from the manifest), along with extra entries for the additional files that we know to be okay (e.g. .travis.yml).  And both the update command and this test should use that method.

Reply:
Understood! Looking at update_util.py I can see how that additional function makes sense there, plus we could unit test the function directly, in addition to this integ test.